src/main/java/data/CameraTimeline.java

```diff
@@ -93,17 +93,31 @@ public CameraTimeline(Camera camera, String description, ScriptingProject projec
                     added = true;
                 }
             }
-            if (checkOverlap(shot, other, camera.getMovementMargin())) {
-                result.add(other);
-            }
         }
         if (!added) {
             shots.add(shot);
         }
+        return getOverlappingShots(shot);
+    }
+
+    /**
+     * Get the list of shots overlapping with the given shots.
+     * @param shot - the shot to check with
+     * @return - only the shot when no overlap, list of overlapping shots otherwise
+     */
+    public ArrayList<CameraShot> getOverlappingShots(CameraShot shot) {
+        ArrayList<CameraShot> result = new ArrayList<>();
+
+        // check for overlapping shots
+        result.addAll(shots.stream()
+                .filter(other -> shot != other && checkOverlap(shot,
+                        other, camera.getMovementMargin()))
```

Comment:
Move the checkOverlap filter to its own filter. Lends readability & doesn't impact performance